feat: add deny-label-ns flag which supports wildcard (#1051)
kubernetes-sigs/node-feature-discovery: docs/reference/master-commandline-reference.md

nfd-master -extra-label-ns=vendor-1.com,vendor-2.io
```

### -deny-label-ns

The `-deny-label-ns` flag specifies a comma-separated list of excluded
label namespaces. By default, nfd-master allows creating labels in all
namespaces, excluding `*.kubernetes.io` namespace and its sub-namespaces
(e.g. `sub.ns.kubernetes.io`). However, you should note that the
`*.kubernetes.io` label is forcibly denied.
e.g: `nfd-master -deny-label-ns=""` would still deny `*.kubernetes.io`

> This option can be used to exclude some vendors or application specific namespaces.

Reviewer: Format to 80 columns 

```suggestion
This option can be used to exclude some vendors or application specific 
namespaces.
```

> Default: *\*.kubernetes.io*

Reviewer 2: With this particular comment I meant

```suggestion
Default: *empty*
```

...as `Default: *\*.kubernetes.io*` would suggest (to me) that you can remove '*kubernetes.io` e.g.with `-deny-label-ns=foo.bar`. *kubernetes.io is always denied irrespective what is specified with this flag

Example:

```bash

> nfd-master -deny-label-ns=kubernetes.io,vendor-2.io

Reviewer 2: A bit misleading example as kubernetes.io should be always forcibly denied anyway

```suggestion
nfd-master -deny-label-ns=*.vendor.com,vendor-2.io
```

nfd-master -extra-label-ns=vendor-1.com,vendor-2.io
```

### -deny-label-ns

The `-deny-label-ns` flag specifies a comma-separated list of excluded
label namespaces. By default, nfd-master allows creating labels in all
namespaces, excluding `kubernetes.io` namespace and its sub-namespaces

> (e.g. `*.kubernetes.io`). However, you should note that the
> `kubernetes.io` namespace is forcibly denied.
> e.g: `nfd-master -deny-label-ns=""` would still deny `*.kubernetes.io`

Reviewer 2: Still bike-shedding on this, I'd suggest to adjust the wording a bit

```suggestion
(i.e. `*.kubernetes.io`). However, you should note that
`kubernetes.io` and its sub-namespaces are always denied.
For example, `nfd-master -deny-label-ns=""` would still disallow
`kubernetes.io` and `*.kubernetes.io`.
```
